pkg/resources/grant_application_role.go

```diff
@@ -39,38 +40,44 @@ var grantApplicationRoleSchema = map[string]*schema.Schema{
 		ForceNew:         true,
 		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
 		ExactlyOneOf: []string{
-			"parent_role_name",
+			"parent_account_role_name",
 			"application_name",
 		},
 	},
 }
 
 func GrantApplicationRole() *schema.Resource {
 	return &schema.Resource{
-		Create: CreateGrantApplicationRole,
-		Read:   ReadGrantApplicationRole,
-		Delete: DeleteGrantApplicationRole,
-		Schema: grantApplicationRoleSchema,
+		CreateContext: CreateContextGrantApplicationRole,
+		ReadContext:   ReadContextGrantApplicationRole,
+		DeleteContext: DeleteContextGrantApplicationRole,
+		Schema:        grantApplicationRoleSchema,
 		Importer: &schema.ResourceImporter{
 			StateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
 				parts := strings.Split(d.Id(), helpers.IDDelimiter)
 				if len(parts) != 3 {
-					return nil, fmt.Errorf("invalid ID specified: %v, expected <name>|<object_type>|<target_identifier>", d.Id())
+					return nil, fmt.Errorf("invalid ID specified: %v, expected <application_role_name>|<object_type>|<target_identifier>", d.Id())
 				}
-				if err := d.Set("name", parts[0]); err != nil {
+				if err := d.Set("application_role_name", parts[0]); err != nil {
 					return nil, err
 				}
 				switch parts[1] {
-				case "ROLE":
-					if err := d.Set("parent_role_name", strings.Trim(parts[2], "\"")); err != nil {
+				case "ACCOUNT_ROLE":
+					p := parts[2]
+					p = strings.Trim(p, "\"")
+					if err := d.Set("parent_account_role_name", p); err != nil {
 						return nil, err
 					}
 				case "APPLICATION":
-					if err := d.Set("application_name", strings.Trim(parts[2], "\"")); err != nil {
+					p := parts[2]
+					if strings.HasPrefix(p, "\"\"") && strings.HasSuffix(p, "\"\"") {
+						p = strings.Trim(p, "\"")
+					}
+					if err := d.Set("application_name", p); err != nil {
```

Comment:
Not sure why now the account role is different than the application. I thought the quoted identifier would be better for being consistent with other id's, so maybe doing this would be enough: `if err := d.Set("application_name", sdk.NewAccountObjectIdentifier(parts[2]).FullyQualifiedName()); err != nil {` in both cases.

Reply:
Fixed.